.github/copilot-instructions.md

```diff
@@ -12,211 +16,62 @@ This repository contains **Azure Verified Modules (AVM)** for Bicep - the offici
+### Use Fetch Tool (When Bicep VS Code Extension Tools Are Not Available)
 
-# Run validation test - dry-run (RECOMMENDED)
-Test-ModuleLocally -TemplateFilePath './avm/res/{service}/{resource}' -ValidationTest -ValidateOrDeployParameters @{subscriptionId='xxx'; managementGroupId='xxx'}
-```
+For the list of resource providers or resource types available, latest API versions, and Bicep schema for specific resources, use the `microsoft_docs_fetch` tool to fetch information from Azure Resource Reference page, available at **https://learn.microsoft.com/en-us/azure/templates/**.
 
-## Local Validation
+#### Fetching Bicep Schemas
 
-Use `utilities/tools/Test-ModuleLocally.ps1` for local testing:
-```powershell
-# Load the script (dot-source)
-. './utilities/tools/Test-ModuleLocally.ps1'
+Use the `fetch` tool to get the Bicep schema for specific resources - use this to find the latest available API version:
 
-# Run Pester tests only
-Test-ModuleLocally -TemplateFilePath './avm/res/storage/storage-account' -PesterTest
+- **URL for specific version**: `https://learn.microsoft.com/azure/templates/{resourceType}/{resourceName}?pivots=deployment-language-bicep`
+- **Example**: `https://learn.microsoft.com/azure/templates/Microsoft.Storage/storageAccounts?pivots=deployment-language-bicep`
 
-# Run validation test (dry-run)
-Test-ModuleLocally -TemplateFilePath './avm/res/storage/storage-account' -ValidationTest -ValidateOrDeployParameters @{subscriptionId='xxx'; managementGroupId='xxx'}
-```
-**Critical**: Use unique `namePrefix` values to avoid token replacement conflicts with module content.
-
-## Common Patterns
-
-### Storage Account Module
-```bicep
-module storageAccount 'br/public:avm/res/storage/storage-account:0.9.1' = {
-  name: 'storageAccountDeployment'
-  params: {
-    name: 'mystorageaccount${uniqueString(resourceGroup().id)}'
-    location: location
-    skuName: 'Standard_LRS'
-    enableTelemetry: true
-  }
-}
-```
+#### Fetching Bicep Schemas with API Version
 
-### Virtual Network Module
-```bicep
-module virtualNetwork 'br/public:avm/res/network/virtual-network:0.8.0' = {
-  name: 'virtualNetworkDeployment'
-  params: {
-    name: 'myVnet'
-    location: location
-    addressPrefixes: ['10.0.0.0/16']
-    enableTelemetry: true
-  }
-}
-```
+Use the `fetch` tool to get the Bicep schema for specific resources and explicit API version:
 
-### Key Vault Module
-```bicep
-module keyVault 'br/public:avm/res/key-vault/vault:0.7.1' = {
-  name: 'keyVaultDeployment'
-  params: {
-    name: 'myKeyVault${uniqueString(resourceGroup().id)}'
-    location: location
-    enableTelemetry: true
-  }
-}
-```
+- **URL for specific version**: `https://learn.microsoft.com/azure/templates/{resourceType}/{apiVersion}/{resourceName}`
+- **Example**: `https://learn.microsoft.com/azure/templates/Microsoft.Storage/2023-01-01/storageAccounts?pivots=deployment-language-bicep`
 
 ## Fetching Schemas, API versions and existing Published AVM Modules
```

Comment:
I'll need some help with this. I wasn't able to judge whether this is all correct, so I just kept what we already had.